packages/frontend/web/components/AdSlot.tsx

```diff
@@ -0,0 +1,66 @@
+// tslint:disable:react-no-dangerous-html
+
+import React from 'react';
+
+export const AdSlot: React.FC<{
+    name: string;
+    adTypes: string[];
+    sizeMapping: {
+        [key: string]: string[];
+    };
+    showLabel?: boolean;
+    refresh?: boolean;
+    outOfPage?: boolean;
+    optId?: string;
+    optClassNames?: string[];
+}> = ({
+    name,
+    adTypes,
+    sizeMapping,
+    showLabel = true,
+    refresh = true,
+    outOfPage = false,
+    optId,
+    optClassNames,
+}) => {
+    const dataSizeMappings = Object.keys(sizeMapping).reduce(
```

Comment:
Wondering if this would be useful to pull out and run tests against, it's a bit gnarly to reason with

Reply:
+1